### DISCUSSION TITLE
Plugin Development Tutorial for NetBox v3.2

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#8901] [General]
I'm excited to announce that I recently completed a [plugin development tutorial](https://github.com/netbox-community/netbox-plugin-tutorial) for the upcoming NetBox v3.2 release! As you may know, the v3.2 release _greatly_ extends NetBox's plugin framework and makes plugin development a great deal more accessible.

The tutorial walks through creating a custom plugin for managing simple access lists from scratch. It comprises ten discrete steps, each focusing on a particular component of the plugins framework. It also features a companion repo housing a snapshot of the completed code corresponding to each step. This enables the reader to easily jump in at any point along the way.

* [Plugin Development Tutorial](https://github.com/netbox-community/netbox-plugin-tutorial)
* [Demo Plugin](https://github.com/netbox-community/netbox-plugin-demo)

Please bear in mind that this tutorial was written for NetBox v3.2, so you'll need to install the most recent [beta release](https://github.com/netbox-community/netbox/releases/tag/v3.2-beta2) locally to code along. And although I've made every effort to ensure a smooth experience, there may still be issues. Your feedback is greatly appreciated!

My hope is that this tutorial will lower the barrier to entry for people to develop their own plugins. Hopefully you'll be inspired to come up with something of your own!

![netbox-plugin-demo](https://user-images.githubusercontent.com/13487278/158883386-0f803366-85b2-415a-b093-e76e8733c699.png)

**Comment by 3bruno** [comment]
Hi Jeremy. The tutorial is already complete?  

I'm following the netbox-plugin-tutorial and I cannot testing the view in the step 5. I'm not modify any files, just cloning the repository and follow the tutorial and getting a "NoReverseMatch at /plugins/access-lists/access-lists/" error.

The template from objectview is working fine like the image from this post, but the object_list (access list list view) is not working.

Is it working in your environment?

**Reply by jeremystretch** [reply]
Hey, thanks for trying it out! Would you mind opening an issue [under the tutorial repo](https://github.com/netbox-community/netbox-plugin-tutorial/issues/new)? Trying to keep bug reports centralized there. Thanks!

**Comment by iunderwood** [comment]
I read through this over the weekend, and it's one of the best module tutorials i have seen in a long time.

If work lets up a little this week (lol), I'm going to try following along.  I've got a mind to write something focused more on VPN management, but the data structure starting out would be very similar to the ACL tutorial you have described.
